velox/common/caching/SsdFile.cpp

```diff
@@ -438,6 +450,26 @@ void SsdFile::write(std::vector<CachePin>& pins) {
   }
 }
 
+bool SsdFile::write(
+    std::vector<iovec>& iovecs,
+    uint64_t offset,
+    uint64_t expectedWrittenBytes) {
+  const auto rc = folly::pwritev(fd_, iovecs.data(), iovecs.size(), offset);
+  if (rc != expectedWrittenBytes) {
+    VELOX_SSD_CACHE_LOG(ERROR)
+        << "Failed to write to SSD, file name: " << fileName_ << ", fd: " << fd_
+        << ", size: " << iovecs.size() << ", offset: " << offset
+        << ", error code: " << errno
+        << ", error string: " << folly::errnoStr(errno);
+    ++stats_.writeSsdErrors;
+    // If write fails, we return without adding the pins to the cache. The
+    // entries are unchanged.
+    return false;
+  }
+  iovecs.clear();
```

Comment:
Let the caller to clear iovecs and then pass const reference (const std::vector<iovec>& iovecs) to write function?